### PR TITLE
DOCSP-42666: serialization

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-42666>
Staging - https://preview-mongodbrustagir.gatsbyjs.io/kotlin-sync/DOCSP-42666-serialization/data-formats/serialization/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/data-formats/serialization.txt [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+- All {+language+} types that are supported by the ``kotlinx.serialization`` library
+- All :manual:`BSON types </reference/bson-types>`
+
+.. _kotlin-sync-add-serialization:
+
+Add the Serialization Dependencies to Your Project
+--------------------------------------------------
+
+You must install the official {+language+} serialization library,
+``kotlinx.serialization`` to serialize and deserialize data in your
```
Comment (mcmorisi) [inline source/data-formats/serialization.txt:62]:
```suggestion
``kotlinx.serialization``, to serialize and deserialize data in your
```

File: source/data-formats/serialization.txt [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+Use the ``BsonConfiguration`` class to define the configuration,
+which can include whether to encode defaults, encode nulls, or define
+class discriminators.
+
+To create a custom codec, your project must have the ``bson-kotlinx``
+dependency. See the :ref:`kotlin-sync-add-serialization` section of this
+guide for installation instructions.
+
+Then, you can define your codec by using the `KotlinSerializerCodec.create()
```
Comment (mcmorisi) [inline source/data-formats/serialization.txt:212]:
```suggestion
You can define your codec by using the `KotlinSerializerCodec.create()
```

File: source/data-formats/serialization.txt [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+<{+java-api+}/apidocs/bson-kotlinx/bson-kotlinx/org.bson.codecs.kotlinx/-kotlin-serializer-codec/-companion/index.html>`__
+method and add it to the registry.
```
Comment (mcmorisi) [inline source/data-formats/serialization.txt:214]:
```suggestion
method and adding it to the registry.
```

File: source/includes/data-formats/serialization.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    collection.insertOne(teacherDoc)
+    collection.insertOne(studentDoc)
+
+    println("Retrieving by using data classes")
+    val resultTeacher = collection.withDocumentClass<Teacher>()
+        .find(Filters.exists("department"))
+        .firstOrNull()
+    println(resultTeacher)
+
+    val resultStudent =collection.withDocumentClass<Student>()
```
Comment (mcmorisi) [inline source/includes/data-formats/serialization.kt:111]:
```suggestion
    val resultStudent = collection.withDocumentClass<Student>()
```

**Review by mcmorisi — APPROVED** [review]
LGTM with some tiny nits!

**Review by rozza — CHANGES_REQUESTED** [review]
Recommend using the https://www.mongodb.com/docs/drivers/kotlin/coroutine/current/fundamentals/data-formats/serialization/ page wholesale.

Its important to have the differentiation between kotlinx serialization and using codecs. Which appears to be missing from this page.

**Pull request merged**